### PR TITLE
 Use default template manager when the builder does not have one

### PR DESCRIPTION
**Pull request by bosonogi**
Instead of ignoring the `datatemplate` directive when the builder does not use a template manager, initialize a `sphinx.jinja2glue.BuiltinTemplateLoader` template manager and use that.

Used successfully for _latex_ and _confluence_ targets.

Also, added an empty [merge_domaindata](https://www.sphinx-doc.org/en/master/extdev/domainapi.html#sphinx.domains.Domain.merge_domaindata) method to the domain for parallel builds to continue working.

### CODE REVIEW

File: sphinxcontrib/datatemplates/directive.py [hunk 1]
```diff
@@ -16,6 +16,25 @@
...
+def _templates(builder):
+    global _default_templates
+
+    # Some builders have no templates manager at all, and some
+    # have the attribute set to None.
+    templates = getattr(builder, 'templates', None)
+
+    if not templates:
+        if not _default_templates:
+            from sphinx.jinja2glue import BuiltinTemplateLoader
```
Comment (dhellmann) [inline sphinxcontrib/datatemplates/directive.py]: I generally like to have all of the imports at the top of the file, unless doing that is going to introduce some sort of circular import case. It's not a big deal, and you could make the change by pushing another patch to this PR or in a separate PR, if you prefer.

**Review by dhellmann — COMMENTED** [review]
One nit picky review comment. I'm comfortable taking this as-is but wanted to see what you thought before I did that.

File: sphinxcontrib/datatemplates/directive.py (hunk 1 above)
Comment (bosonogi) [reply sphinxcontrib/datatemplates/directive.py]: @dhellmann It is an artifact from the initial approach that I abandoned (where a separate module was used to initialize a template manager and it was imported only if needed).

I'll clean this up a bit and push the changes here.

Comment (dhellmann) [reply sphinxcontrib/datatemplates/directive.py]: Looks good, thanks!
